apps/prairielearn/python/prairielearn.py

```diff
@@ -352,7 +357,78 @@ def to_json(v, *, df_encoding_version=1, np_encoding_version=1):
         return v
 
 
-def from_json(v):
+class _JSONSerializedComplex(TypedDict):
+    _type: Literal["complex"]
+    _value: dict[str, float]
```

Comment:
This could be slightly more specific, but I didn't see a lot of value in doing so.

Reply:
A lot of these are probably overkill given that the dictionaries are checked for entries being present in the function itself. Does it make a difference if we just use a single typeddict with a unioned literal for the _type?

Reply:
I was thinking that these types might have more value down the road with to_json/from_json to keep some typed-ness.

Reply:
Curious what @nwalters512 thinks. If this winds up being needed, we can always add later, but for now it makes the file a little bloated.

Reply:
Sounds like the types could be split into a separate file if we are worried about file size.

Reply:
I agree that this is entirely overkill at this point. I also don't think we can use these in the future for "more typedness" in the JSON functions since we're always crossing process boundaries there and we can't rely on data flowing into `from_json` to have any specific shape.

Reply:
Changed in a0fc3b725a9104706c99dea8fbbdc23ea36b757f.

Reply:
@eliotwrobson I did some overloads, so that certain callsites can guarantee their return type is of `_JSONSerializedType`